### PR TITLE
added checks on DR page for new ARD functionality

### PR DESCRIPTION
**Pull request by amoriartyCH**
-Added validation to check whether appointment or resignation dates are invalid after changing the accounting reference date.

-Added and fixed unit tests to cover new functionality.

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/web/accounts/validation/smallfull/DirectorValidator.java [hunk 1]
```diff
@@ -1,18 +1,32 @@
...
+import uk.gov.companieshouse.web.accounts.model.directorsreport.Director;
 import uk.gov.companieshouse.web.accounts.model.directorsreport.DirectorToAdd;
+import uk.gov.companieshouse.web.accounts.service.smallfull.SmallFullService;
 import uk.gov.companieshouse.web.accounts.validation.ValidationError;
 
 @Component
 public class DirectorValidator {
 
+    @Autowired
+    SmallFullService smallFullService;
```
Comment (BSaunders95) [inline src/main/java/uk/gov/companieshouse/web/accounts/validation/smallfull/DirectorValidator.java:25]: Access modifiers ;)

File: src/main/java/uk/gov/companieshouse/web/accounts/validation/smallfull/DirectorValidator.java [hunk 2]
```diff
@@ -81,6 +99,46 @@
...
+        }
+
         return validationErrors;
     }
+
+    private boolean isValidAppointmentOrResignationDate(Director director, LocalDate periodStartOn, LocalDate periodEndOn) {
+        boolean isValid = true;
+
+        if (director.getResignationDate() != null) {
+            if (director.getResignationDate().isBefore(periodStartOn) || director.getResignationDate().isAfter(periodEndOn)) {
```
Comment (BSaunders95) [inline src/main/java/uk/gov/companieshouse/web/accounts/validation/smallfull/DirectorValidator.java:132]: Consider combining this if with the wrapping if?

File: src/test/java/uk/gov/companieshouse/web/accounts/validation/smallfull/DirectorValidatorTest.java [hunk 3]
```diff
@@ -3,24 +3,44 @@
...
+import uk.gov.companieshouse.web.accounts.service.smallfull.SmallFullService;
 import uk.gov.companieshouse.web.accounts.validation.ValidationError;
 
 @ExtendWith(MockitoExtension.class)
 @TestInstance(TestInstance.Lifecycle.PER_CLASS)
 public class DirectorValidatorTest {
 
-    private DirectorValidator validator = new DirectorValidator();
+    @Mock
+    SmallFullService smallFullService;
```
Comment (BSaunders95) [inline src/test/java/uk/gov/companieshouse/web/accounts/validation/smallfull/DirectorValidatorTest.java:34]: Access modifiers here, too

File: src/main/java/uk/gov/companieshouse/web/accounts/validation/smallfull/DirectorValidator.java (hunk 1 above)
Comment (amoriartyCH) [reply src/main/java/uk/gov/companieshouse/web/accounts/validation/smallfull/DirectorValidator.java]: Fixed hahaha

File: src/main/java/uk/gov/companieshouse/web/accounts/validation/smallfull/DirectorValidator.java (hunk 2 above)
Comment (amoriartyCH) [reply src/main/java/uk/gov/companieshouse/web/accounts/validation/smallfull/DirectorValidator.java]: Took this advice, new code pushed contains changes.

**Pull request merged**